macOSUpgrade.sh

```diff
@@ -153,10 +153,38 @@ osinstallersetupdDaemonSettingsFilePath="/Library/LaunchDaemons/com.jamfps.clean
 ##Launch agent settings for filevault authenticated reboots
 osinstallersetupdAgentSettingsFilePath="/Library/LaunchAgents/com.apple.install.osinstallersetupd.plist"
 
+##Amount of time (in seconds) to allow a user to connect to AC power before moving on
+##If null or 0, then the user will not have the opportunity to connect to AC power
+acPowerWaitTimer="0"
+
+##Icon to display during the AC Power warning
+warnIcon="/System/Library/CoreServices/CoreTypes.bundle/Contents/Resources/AlertCautionIcon.icns"
+
+##Icon to display when errors are found
+errorIcon="/System/Library/CoreServices/CoreTypes.bundle/Contents/Resources/AlertStopIcon.icns"
+
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 # FUNCTIONS
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
+wait_for_ac_power() {
+    # Loop for "acPowerWaitTimer" seconds until either AC Power is detected or the timer is up
+    sysRequirementErrors=()
```

Comment:
In your use case this variable is used as a global rather than local variable.
If you want to use it as a global variable, please define and initialize it at the beginning of script.

Reply:
Declared array at top of script.